org.scala-ide.sdt.core.tests/src/org/scalaide/core/sbtbuilder/ScalaProjectDependedOnJavaProjectTest.scala

```diff
@@ -0,0 +1,79 @@
+package org.scalaide.core.sbtbuilder
+
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.runtime.IPath
+import org.eclipse.jdt.core.IJavaProject
+import org.eclipse.jdt.core.JavaCore
+import org.junit.AfterClass
+import org.junit.Assert
+import org.junit.BeforeClass
+import org.junit.Test
+import org.scalaide.core.IScalaProject
+import org.scalaide.core.SdtConstants
+import org.scalaide.core.testsetup.Bdd
+import org.scalaide.core.testsetup.Bdd.jProjectToIProject
+import org.scalaide.core.testsetup.Bdd.sProjectToIProject
+import org.scalaide.core.testsetup.Before
+import org.scalaide.core.testsetup.SDTTestUtils
+import org.scalaide.core.testsetup.SDTTestUtils.SrcPathOutputEntry
+import org.scalaide.core.testsetup.SDTTestUtils.addToClasspath
+import org.scalaide.core.testsetup.SDTTestUtils.createJavaProjectInWorkspace
+import org.scalaide.core.testsetup.SDTTestUtils.createProjectInWorkspace
+import org.scalaide.util.eclipse.EclipseUtils
+
+import ScalaProjectDependedOnJavaProjectTest.projectJ
+import ScalaProjectDependedOnJavaProjectTest.projectS
+
+object ScalaProjectDependedOnJavaProjectTest extends Before {
+  import org.scalaide.core.testsetup.SDTTestUtils._
+  val projectJName = "scalaDependedOnJavaJ"
+  val projectSName = "scalaDependedOnJavaS"
+  var projectJ: IJavaProject = _
+  var projectS: IScalaProject = _
+  val bundleName = "org.scala-ide.sdt.core.tests"
+
+  private val withSrcOutputStructure: SrcPathOutputEntry = (project, jProject) => {
```

Comment:
I would prefer def with explicit parameter types but I guess it depends on tastes ;)